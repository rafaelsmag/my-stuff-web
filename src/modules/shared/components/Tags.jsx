import React from 'react'
import { arrayOf, shape, func, string } from 'prop-types'
import { Icon } from 'semantic-ui-react'
import { Div } from 'glamorous'
import { Metrics, Fonts, Colors } from '../../../config/constants'

const renderTags = (tags, onClick) => {
  return tags.map((tag) => (
    <Div css={styles.tagName} onClick={() => onClick(tag)}>{tag.name}</Div>
  ))
}

const Tags = ({ tags, onClickTag }) => (
  <Div css={styles.container}>
    <Div>
      <Icon name="tags" disabled size='small'/>
    </Div>
    {renderTags(tags, onClickTag)}
  </Div>
)

Tags.defaultProps = {
  onClickTag: () => {}
}

Tags.propTypes = {
  tags: arrayOf(shape({
    name: string.isRequired,
    _id: string
  })).isRequired,
  onClickTag: func
}

const styles = {
  container: {
    display: 'flex',
    flex: '1 1 auto',
    marginBottom: Metrics.xsSpacing,
    flexWrap: 'wrap'
  },
  tagName: {
    margin: `0 ${Metrics.xxsSpacing} ${Metrics.xxsSpacing} 0`,
    padding: Metrics.xxsSpacing,
    // backgroundColor: Colors.lightBackground,
    borderRadius: Fonts.size.tiny,
    fontSize: Fonts.size.tiny,
    lineHeight: Fonts.size.tiny,
    color: Colors.primary,
    // textTransform: 'uppercase',
    fontWeight: '600'
  }
}

export default Tags
