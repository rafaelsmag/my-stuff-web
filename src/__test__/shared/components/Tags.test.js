import React from 'react'
import { shallow } from 'enzyme'

import Tags, { renderTags } from '../../../modules/shared/components/Tags'

describe('Tags Component', () => {
  const mockOnClickTag = jest.fn()
  const props = {
    tags: [{ name: 'Tag 1', _id: '1' }, { name: 'Tag 2', _id: '2' }],
    onClickTag: mockOnClickTag
  }
  const tags = shallow(<Tags {...props} />)

  it('should render properly', () => {
    expect(tags).toMatchSnapshot()
  })

  it('should render 2 tags', () => {
    expect(tags.children().find({ id: props.tags[0]._id }).exists()).toBe(true)
    expect(tags.children().find({ id: props.tags[1]._id }).exists()).toBe(true)
  })

  describe('when calling `renderTags` function', () => {
    const resultRenderTags = renderTags(props.tags, props.onClickTag)
    it('should return an array with 2 itens', () => {
      expect(resultRenderTags).toHaveLength(2)
    })
  })

  describe('when clicking on a `Tag`', () => {
    it('should call `onClickTag` function', () => {
      tags.children().find({ id: props.tags[0]._id }).simulate('click')
      expect(mockOnClickTag).toHaveBeenCalledWith(props.tags[0])
    })
  })
})
