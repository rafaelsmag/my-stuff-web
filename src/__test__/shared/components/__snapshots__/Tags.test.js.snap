// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags Component should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Tags
    onClickTag={[Function]}
    tags={
        Array [
            Object {
              "_id": "1",
              "name": "Tag 1",
            },
            Object {
              "_id": "2",
              "name": "Tag 2",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous.Div>
          <Icon
                    as="i"
                    disabled={true}
                    name="tags"
                    size="small"
          />
</glamorous.Div>,
        Array [
          <glamorous.Div
            css={
                        Object {
                                    "borderRadius": "0.8em",
                                    "color": "#15c850",
                                    "fontSize": "0.8em",
                                    "fontWeight": "600",
                                    "margin": "0 4px 4px 4px",
                                  }
            }
            id="1"
            onClick={[Function]}
>
            Tag 1
</glamorous.Div>,
          <glamorous.Div
            css={
                        Object {
                                    "borderRadius": "0.8em",
                                    "color": "#15c850",
                                    "fontSize": "0.8em",
                                    "fontWeight": "600",
                                    "margin": "0 4px 4px 4px",
                                  }
            }
            id="2"
            onClick={[Function]}
>
            Tag 2
</glamorous.Div>,
        ],
      ],
      "css": Object {
        "display": "flex",
        "flex": "1",
        "flexWrap": "wrap",
        "marginTop": "6px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Icon
            as="i"
            disabled={true}
            name="tags"
            size="small"
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "i",
            "disabled": true,
            "name": "tags",
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "children": "Tag 1",
          "css": Object {
            "borderRadius": "0.8em",
            "color": "#15c850",
            "fontSize": "0.8em",
            "fontWeight": "600",
            "margin": "0 4px 4px 4px",
          },
          "id": "1",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Tag 1",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "children": "Tag 2",
          "css": Object {
            "borderRadius": "0.8em",
            "color": "#15c850",
            "fontSize": "0.8em",
            "fontWeight": "600",
            "margin": "0 4px 4px 4px",
          },
          "id": "2",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Tag 2",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous.Div>
            <Icon
                        as="i"
                        disabled={true}
                        name="tags"
                        size="small"
            />
</glamorous.Div>,
          Array [
            <glamorous.Div
              css={
                            Object {
                                          "borderRadius": "0.8em",
                                          "color": "#15c850",
                                          "fontSize": "0.8em",
                                          "fontWeight": "600",
                                          "margin": "0 4px 4px 4px",
                                        }
              }
              id="1"
              onClick={[Function]}
>
              Tag 1
</glamorous.Div>,
            <glamorous.Div
              css={
                            Object {
                                          "borderRadius": "0.8em",
                                          "color": "#15c850",
                                          "fontSize": "0.8em",
                                          "fontWeight": "600",
                                          "margin": "0 4px 4px 4px",
                                        }
              }
              id="2"
              onClick={[Function]}
>
              Tag 2
</glamorous.Div>,
          ],
        ],
        "css": Object {
          "display": "flex",
          "flex": "1",
          "flexWrap": "wrap",
          "marginTop": "6px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Icon
              as="i"
              disabled={true}
              name="tags"
              size="small"
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "i",
              "disabled": true,
              "name": "tags",
              "size": "small",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "children": "Tag 1",
            "css": Object {
              "borderRadius": "0.8em",
              "color": "#15c850",
              "fontSize": "0.8em",
              "fontWeight": "600",
              "margin": "0 4px 4px 4px",
            },
            "id": "1",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Tag 1",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "children": "Tag 2",
            "css": Object {
              "borderRadius": "0.8em",
              "color": "#15c850",
              "fontSize": "0.8em",
              "fontWeight": "600",
              "margin": "0 4px 4px 4px",
            },
            "id": "2",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Tag 2",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
